hledger/hledger.m4.md

```diff
@@ -4615,6 +4615,15 @@ These are most often [account name](#account-names) substrings:
 
   `not:cur:USD`
 
+When quotes are used to escape spaces and/or special characters, the entire query should be enclosed with quotes as well:
+
+   `"'^expenses\b' 'accounts (payable|receivable)\""`
+
+Note that we use double quotes to escape the query and single quotes to escape the individual parts of the query.
+Escaping parts of the query with double quotes would also be possible, but those quotes would need to be escaped to be part of the query:
+
+   `"'^expenses\b' \"accounts (payable|receivable)\""`
+
```

Comment:
I appreciate the attention to easy migration.

Am I understanding rightly this would break some existing reports involving multi-word queries ? Eg a report like `hledger bal 'my bank'` would now instead be equivalent to our current `hledger bal my bank`, possibly showing different results ?

Reply:
I am frankly scared of complicating quoting, escaping and parsing of the command line any further. I feel we are already pushing the limit, as you can maybe tell from `words'` and `words''` and https://hledger.org/dev/hledger.html#special-characters. If we must break current command line semantics anyway - not saying we will, but if we do - perhaps we should just accept the boolean query terms as arguments without the extra level of quoting ?

Reply:
Awesome @chrislemaire that you're working ln this, thank you! 🎉 

I also find it problematic that queries with accounts that contain sapaces will be broken with this.

How about introducing a new directive for boolean queries (e.g. `query:`) that is parsed normally but then adds to the filtering? With the new proposed expressions one has to use quotes anyway, so just putting a `query:` in front of the potentially elaborate expressions doesn't hurt, but maintains 100% backwards compatibility.

Reply:
I see. I do agree the many quotes would be a bit overwhelming. Some extra quoting will still be required with `expr:`, but at least the rules for that will be consistent with the existing quoting rules!

I changed the PR so that boolean queries need to be prefixed with `expr:` as Simon suggested. The test now passes without having to change it, so that resolves our quoting issue.